lib/collection/src/shards/replica_set/mod.rs

```diff
@@ -297,6 +297,11 @@ impl ShardReplicaSet {
         matches!(*local_read, Some(Shard::Local(_) | Shard::Dummy(_)))
     }
 
+    pub async fn is_queue_proxy_local(&self) -> bool {
```

Comment:
`is_queue_proxy_local` feels wrong, IMO.

```suggestion
    pub async fn is_queue_proxy(&self) -> bool {
```